protocol/xmrtaker/swap_state.go

```diff
@@ -669,7 +661,39 @@ func (s *swapState) lockAsset() (*ethtypes.Receipt, error) {
 		return nil, err
 	}
 
-	log.Infof("locked %s in swap contract, waiting for XMR to be locked", providedAmt.StandardSymbol())
+	log.Infof("locked %s in swap contract, waiting for XMR to be locked", s.providedAmount.StandardSymbol())
+	return receipt, nil
+}
+
+func (s *swapState) lockAndWaitForReceipt(
```

Comment:
The maker function is called `lockFunds`. `lockXMRFunds` and `lockETHFunds` might make it easier to search for the function you want. I think the receipt portion of the function name is implied, as the funds are not locked if the transaction is still in the mempool.

Reply:
this function is called inside the function `lockAsset`, should i rename that to `lockFunds`?

Reply:
This function appears to just be a wrapper around calling `newSwap` directly on the the contract. Looking at the functions below it that do very similar things, we are just calling them the same name as the function on the contract. Maybe calling it `newSwap` would make the code consistent?  Also: The next function down should be renamed to `setReady`. If you look at its documentation, it appears to have been written at a point in time when the contract method was called `ready`.

Reply:
I think renaming it `newSwap` is way too unclear, I don't think these function names need to match the contract functions. I'll rename/update the below function though